src/hotspot/os/linux/cgroupV2Subsystem_linux.cpp

```diff
@@ -315,7 +315,11 @@ char* CgroupV2Controller::construct_path(char* mount_path, const char* cgroup_pa
   stringStream ss;
   ss.print_raw(mount_path);
   if (strcmp(cgroup_path, "/") != 0) {
-    ss.print_raw(cgroup_path);
+    if (strstr(cgroup_path, "../") == nullptr) {
+      ss.print_raw(cgroup_path);
+    } else {
+      log_warning(os, container)("Cgroup cpu/memory controller path includes '../', detected limits won't be accurate");
+    }
```

Comment:
Please move this warning to `CgroupUtil::adjust_controller` and abort the adjustment, we don't need to issue this warning multiple times, and we'd not be able to adjust it to a path that will work. Showing the warning once should be sufficient. We shouldn't see this path in any non-moved scenarios. It would perhaps help if we included some detail why this warning is being shown. I suggest:

```cgroup controller path seems to have moved (includes '.../'), detected limits won't be accurate```

Reply:
Would you recommand also to include the paths in that warning? Something like
```cgroup controller path at '/sys/fs/cgroup' seems to have moved to '../../test', detected limits won't be accurate```
This way it will have all the necessary information to investigate customer cases.

Reply:
Updated the log message. The log example in cg v1:
```
[0.001s][trace][os,container] OSContainer::init: Initializing Container Support
[0.001s][debug][os,container] Detected optional pids controller entry in /proc/cgroups
[0.001s][debug][os,container] Detected cgroups hybrid or legacy hierarchy, using cgroups v1 controllers
[0.001s][warning][os,container] Cgroup memory controller path at '/sys/fs/cgroup/memory' seems to have moved to '/../../test', detected limits won't be accurate
[0.001s][debug  ][os,container] OSContainer::init: is_containerized() = true because all controllers are mounted read-only (container case)
[0.001s][trace  ][os,container] Path to /cpu.cfs_quota_us is /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
[0.001s][trace  ][os,container] CPU Quota is: -1
[0.002s][trace  ][os,container] Path to /cpu.cfs_period_us is /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us
[0.002s][trace  ][os,container] CPU Period is: 100000
[0.002s][trace  ][os,container] OSContainer::active_processor_count: 48
[0.002s][trace  ][os,container] CgroupSubsystem::active_processor_count (cached): 48
[0.002s][trace  ][os,container] total physical memory: 133623721984
[0.002s][trace  ][os,container] Path to /memory.limit_in_bytes is /sys/fs/cgroup/memory/memory.limit_in_bytes
[0.002s][trace  ][os,container] Memory Limit is: 419430400
[0.004s][trace  ][os,container] CgroupSubsystem::active_processor_count (cached): 48
[0.027s][trace  ][os,container] Path to /cpu.cfs_quota_us is /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
[0.027s][trace  ][os,container] CPU Quota is: -1
[0.027s][trace  ][os,container] Path to /cpu.cfs_period_us is /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us
[0.027s][trace  ][os,container] CPU Period is: 100000
[0.027s][trace  ][os,container] OSContainer::active_processor_count: 48
openjdk version "24-internal" 2025-03-18
OpenJDK Runtime Environment (build 24-internal-adhoc.bellsoft.jdk)
OpenJDK 64-Bit Server VM (build 24-internal-adhoc.bellsoft.jdk, mixed mode, sharing)
```

Reply:
@jerboaa Could you please take a look?